### PR TITLE
Multi snfe

### PR DESCRIPTION
**Pull request by keegansmith21**
Made a function in snfe to process multiple pulsars from the same obs. Massive speed increase for find_pulsar_in_obs.py *obsid* --sn_est. Should also speed up singular queries although i haven't tested those. Some makeshift benchmarks using bash time:

=== fpio benchmarks: ===
**Before Upgrade:**
1267283936:
real	10m19.565s
user	8m12.245s
sys	0m32.437s

1264867416:
real	16m8.664s
user	10m17.154s
sys	0m41.584s

**After Upgrade:**
1267283936:
real	1m44.521s
user	1m41.110s
sys	0m29.091s

1264867416:
real	1m53.941s
user	1m55.624s
sys	0m37.103s

### CODE REVIEW

File: scripts/find_pulsar_in_obs.py [hunk 1]
```diff
@@ -113,24 +113,26 @@ def get_psrcat_ra_dec(pulsar_list=None, max_dm=1000., include_dm=False):
...
 
-    params = ['JNAME', 'RAJ', 'DECJ', 'DM']
-    query = psrqpy.QueryATNF(params=params, psrs=pulsar_list, loadfromdb=ATNF_LOC).pandas
+    #params = ['JNAME', 'RAJ', 'DECJ', 'DM']
+    if not query:
+        query = psrqpy.QueryATNF(params = ['PSRJ', 'RAJ', 'DECJ', 'DM'], psrs=pulsar_list, loadfromdb=ATNF_LOC).pandas
 
     pulsar_ra_dec = []
-    for row in query.itertuples():
+    for i, psr in enumerate(query["PSRJ"]):
```
Comment (NickSwainston) [inline scripts/find_pulsar_in_obs.py]: Codacy Issue found: [Unused variable 'psr'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=5199130)

**Comment by NickSwainston** [comment]
That's some excellent work and a great speed up.

Looks like it's breaking a few of the tests (which sometimes makes me happy that we put the time in to make them) so I'll wait for you to fix those. I'd also recommend changing test_est_pulsar_sn() so it uses your quicker new process.

File: tests/test_sn_flux_est.py [hunk 2]
```diff
@@ -71,22 +82,28 @@ def test_find_t_sys_gain():
...
+    md_2 = md_dict[str(obsid_2)][0]
+    full_md_2 = md_dict[str(obsid_2)][1]
+
     q_1 = psrqpy.QueryATNF(psrs=[pulsar_1], loadfromdb=ATNF_LOC)
     q_2 = psrqpy.QueryATNF(psrs=[pulsar_2], loadfromdb=ATNF_LOC)
     test_cases = []
-    test_cases.append((pulsar_1, obsid_1, 1223042887, 600, q_1, md_1,\
+    #test_cases.append((pulsar_1, obsid_1, 1223042887, q_1, md_1, full_md_1,\
+    #            325.16928459135534, 6.5033856918271065, 0.1405071004856564, 0.1065230048864121))
+
```
Comment (NickSwainston) [inline tests/test_sn_flux_est.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=5199130)

File: tests/test_sn_flux_est.py [hunk 3]
```diff
@@ -71,22 +82,28 @@ def test_find_t_sys_gain():
...
     q_1 = psrqpy.QueryATNF(psrs=[pulsar_1], loadfromdb=ATNF_LOC)
     q_2 = psrqpy.QueryATNF(psrs=[pulsar_2], loadfromdb=ATNF_LOC)
     test_cases = []
-    test_cases.append((pulsar_1, obsid_1, 1223042887, 600, q_1, md_1,\
-                325.17339020027805, 6.5034678040055613, 0.12547449408456718, 0.095633857616224477))
-    test_cases.append((pulsar_2, obsid_2, None, None, q_2, md_2,\
-                295.73944858721245, 5.9147889717442492, 0.29127750035795497, 0.049367868667752078))
+    #test_cases.append((pulsar_1, obsid_1, 1223042887, q_1, md_1, full_md_1,\
+    #            325.16928459135534, 6.5033856918271065, 0.1405071004856564, 0.1065230048864121))
+
```
Comment (NickSwainston) [inline tests/test_sn_flux_est.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=5199130)

**Comment by keegansmith21** [comment]
Ready for review @NickSwainston

**Pull request merged**